velox/exec/PartitionedOutputBufferManager.h

```diff
@@ -80,25 +120,38 @@ class DestinationBuffer {
   int64_t sequence_ = 0;
   DataAvailableCallback notify_ = nullptr;
   // The sequence number of the first item to pass to 'notify'.
-  int64_t notifySequence_;
-  uint64_t notifyMaxBytes_;
+  int64_t notifySequence_{0};
+  uint64_t notifyMaxBytes_{0};
 };
 
 class PartitionedOutputBuffer {
  public:
+  enum class Kind {
+    kPartitioned,
+    kBroadcast,
+    kArbitrary,
+  };
+  static std::string kindString(Kind kind);
+
   PartitionedOutputBuffer(
       std::shared_ptr<Task> task,
-      bool broadcast,
+      Kind kind,
       int numDestinations,
       uint32_t numDrivers);
 
+  Kind kind() const {
+    return kind_;
+  }
+
+  void updateOutputBuffers(int numBuffers, bool noMoreBuffers);
```

Comment:
Why do we need both updateOutputBuffers and updateBroadcastOutputBuffers ? This buffer seems to have 'kind' property, hence, I'd imagine single updateOutputBuffers method would be sufficient, no?

Reply:
updateBroadcastOutputBuffers is used by Prestissimo and I plan to deprecate this after Presto cpp code has been updated.